prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -55,17 +56,19 @@
 ALLOW_STOP = False
 
 
-def run_agent(agent: Agent, iterations: int, model: str) -> None:
+def run_agent(agent: Agent, iterations: int, model: SupportedModel) -> None:
     maybe_initialize_long_term_memory()
-    with openai_costs(model) as costs:
+    with openai_costs(
+        model.value if model.is_openai else None
+    ) as costs:  # TODO: Support for Replicate costs (below as well).
```

Comment:
This is needed, I will look into it in next PR.

@coderabbitai Please create an issue for this comment.